config/consensus.go

```diff
@@ -440,8 +445,8 @@ var MaxStateDeltaKeys int
 // any version, used only for decoding purposes. Never decrease this value.
 var MaxLogCalls int
 
-// MaxInnerTransactions is the maximum number of inner transactions that may be created in an app call.
-var MaxInnerTransactions int
+// MaxInnerTransactionsPerDelta is the maximum number of inner transactions in one EvalDelta
```

Comment:
The change here is obvious - but it would be causing a potential issue on the "window" between the time a new binary is deployed and the time the new consensus protocol takes place.
To work around that, I suggest adding a manual testing, which be consensus protocol dependent where we would enforce the old limit on older protocol. This would complement the high-bound we set here for the msgpack decoding.

Reply:
Separately, I think that it's great that you renamed the variable so that it would force compile-time validation.

Reply:
I think it's always safe if a later protocol increases the limit in a way that is only used by that new protocol version, as this does.  During the upgrade window, before the new protocol actually goes into effect and some installations have new code and some have old, they will disagree about the allocbound. But no node will produce transactions that _use_ the new allocbound (because the protocol has not been upgraded on the new nodes).

I'm not sure where we would put the manual testing code you suggest. It seems like it belongs in the unmarshalling code, but that's autogenerated.